beeline/__init__.py

```diff
@@ -577,16 +577,16 @@ def finish_span(span):
         bl.tracer_impl.finish_span(span=span)
 
 
-def propagate_and_start_trace(context, headers):
+def propagate_and_start_trace(context, request):
```

Comment:
Does this change the public API that a user of the beeline would need to change to use?

If so, we may need to add a new function (we can't add an overload because python overloading only working with number of params, not based on types like other languages).